### DISCUSSION TITLE
Date shown in 'Time selector' is shifted one day

### DISCUSSION THREAD

**Discussion by iamsaswata** [TerriaJS/terriajs#6968] [Q&A] [answered]
The date shown on the date time selection panel is shifted one day behind.

For the given example (screenshot), we can see date selected is shown as 29/12/2022. But very surprisingly, the date fetched from geoserver is 30/12/2022. I can also confirm actually it is map of 30/12/2022. Configuration is also shown in the given screenshot. 

Any help fixing this is highly appreciated.

<img width="949" alt="terria_issue_datetime_selector" src="https://github.com/TerriaJS/terriajs/assets/15724605/ccd143cd-cf3f-4777-8a6b-cd0915dc7d82">

**Comment by nf-s** [comment]
Hi @iamsaswata 

The dates in Terria will be in your local timezone. Can you please confirm what timezone your in?

Thanks,
Nick

**Comment by iamsaswata** [comment]
Hi @nf-s 
Thanks for looking into this.

My timezone is Pacific Daylight Time (GMT-7) zone.

**Comment by nf-s** [comment, marked as answer]
Hi @iamsaswata

The times provided by your GeoServer are most likely in UTC, but Terria will render dates in your local timezone - this is the default behaviour for handling dates in web browsers. This is why you are seeing a different date between the date/time selector - and the `time` value sent to your GeoServer.

If you want to display UTC times in the Terria UI - you can use the `dateFormat` configuration option.

For example - if you put `UTC:` before your format string, then the date will be shown in UTC instead of your local timezone

```json
"dateFormat": "UTC:dd/mm/yyyy"
```

Let me know how you go!

Thanks 🙂

**Comment by JonDHo** [comment]
This is an issue for all timezones which are behind UTC (and therefore all of the Americas). The solution identified above (using `"dateFormat": "UTC:dd/mm/yyyy"`) will work to force the UI to show the correct date as it forces the UI to format the date to UTC. 

However, this is still an issue if you enable the WMS time filter tool by setting `timeFilterPropertyName`. In the background, the date list for the time filter is converted to non-UTC dates (and therefore are 1 day behind). The data coming from the WMS service has the correct dates, but Terria adjusts the list automatically.

**Comment by nf-s** [comment]
Hi @JonDHo are you able to share your Terria JSON configuration with us? Then we can recreate the problem from our side.
Generally speaking, Terria doesn’t do any processing on dates coming from WMS - so I’m not sure how these values could be incorrect.
As for presenting dates in the UI, web browsers will show dates in their local timezone by default. Using the `dateFormat` parameter can force the UI to show dates in UTC.
The dates you see in the UI are not the dates sent to WMS - we will only send back date strings that are specified in `GetCapabilities` (or `GetFeatureInfo` in the case of `timeFilterPropertyName`)

**Comment by JonDHo** [comment]
Following on from the discussion on Slack, I believe that my issue is with datacube-ows, not Terria, but will link back here if anything is needed on this end.

**Comment by nf-s** [comment]
I am going to convert this into a GitHub discussion - we can create more issues as needed if we come across anything

**Comment by jessjaco** [comment]
Can someone point me to where the `timeFormat` option is described? I don't see it in the docs.

**Reply by JonDHo** [reply]
The `dateFormat` (not `timeFormat`) option is available in various catalog item types (not all), for example in a `WebMapServiceCatalogItem` - https://docs.terria.io/guide/connecting-to-data/catalog-type-details/wms/#:~:text=dateFormat

**Reply by jessjaco** [reply]
Apologies and thank you.
